sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/SecretClientImpl.java

```diff
@@ -617,6 +631,35 @@ private Function<PollingContext<DeletedSecret>, Mono<PollResponse<DeletedSecret>
                     pollingContext.getLatestResponse().getValue()));
     }
 
+    /**
+     * Polling operation to poll on the delete secret operation status.
+     */
+    private Function<PollingContext<DeletedSecret>, PollResponse<DeletedSecret>> createPollOperation(String keyName,
+                                                                                                     Context context) {
+
+        return pollingContext -> {
+            try {
+                Context contextToUse = context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE);
+                contextToUse = enableSyncRestProxy(contextToUse);
+                Response<DeletedSecret> deletedSecretResponse = service.getDeletedSecretPoller(vaultUrl, keyName,
+                    secretServiceVersion.getVersion(), ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE, contextToUse);
+                if (deletedSecretResponse.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+                    return new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS,
+                        pollingContext.getLatestResponse().getValue());
+                }
+
+                return new PollResponse<>(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED,
+                    deletedSecretResponse.getValue());
+            } catch (Exception e) {
+                // This means either vault has soft-delete disabled or permission is not granted for the get deleted key
+                // operation. In both cases deletion operation was successful when activation operation succeeded before
+                // reaching here.
+                return new PollResponse<>(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED,
+                    pollingContext.getLatestResponse().getValue());
+            }
```

Comment:
Should this be scoped to only those scenarios where an error is expected? We should only catch HttpResponseException and check for HTTP status code or error code to validate that it's because of either of the reasons stated in the comment.